daemon/src/housekeeping.rs

```diff
@@ -54,5 +54,13 @@ pub async fn rebroadcast_transactions(
         tracing::info!("Commit transaction published on chain: {}", txid);
     }
 
+    for cfd in cfds.iter().filter(|cfd| Cfd::is_pending_cet(cfd)) {
+        // Double question-mark OK because if we are in PendingCet we must have been Ready before
```

Comment:
Take a note of https://lexi-lambda.github.io/blog/2019/11/05/parse-don-t-validate/ when you have a couple of minutes.

You can avoid such "problems" by combining the check with a type transition. If `is_pending_cet` would not just return you a `bool` but an `Option<>`, you could use `filter_map` here and automatically iterate over only the pending CETs.

The blogpost is relevant because what you are doing here is essentially _parsing_ the pending CETs out of a list of CFDs.